Sources/NIO/Selector.swift

```diff
@@ -389,11 +421,22 @@ final class Selector<R: Registration> {
             default:
                 // If the registration is not in the Map anymore we deregistered it during the processing of whenReady(...). In this case just skip it.
                 if let registration = registrations[Int(ev.data.fd)] {
-                    try body(
-                        SelectorEvent(
-                            readable: (ev.events & Epoll.EPOLLIN.rawValue) != 0 || (ev.events & Epoll.EPOLLERR.rawValue) != 0 || (ev.events & Epoll.EPOLLRDHUP.rawValue) != 0,
-                            writable: (ev.events & Epoll.EPOLLOUT.rawValue) != 0 || (ev.events & Epoll.EPOLLERR.rawValue) != 0 || (ev.events & Epoll.EPOLLRDHUP.rawValue) != 0,
-                            registration: registration))
+                    var selectorEvent: SelectorEventSet = ._none
+                    if ev.events & Epoll.EPOLLIN.rawValue != 0 {
+                        selectorEvent.formUnion(.read)
+                    }
+                    if ev.events & Epoll.EPOLLOUT.rawValue != 0 {
+                        selectorEvent.formUnion(.write)
+                    }
+                    if ev.events & Epoll.EPOLLRDHUP.rawValue != 0 {
+                        selectorEvent.formUnion(.readEOF)
+                    }
+                    if ev.events & Epoll.EPOLLHUP.rawValue != 0 || ev.events & Epoll.EPOLLERR.rawValue != 0 {
+                        selectorEvent.formUnion(.reset)
+                    }
```

Comment:
It'd be nice to bury this in an extension on the type of `ev.events`, I think.

Reply:
I still think this.

Reply:
Though actually I might think it should be an extension containing an initializer for `SelectorEventSet`, actually.